### PR TITLE
polish(rjy): polish the comments of collate_fn/profiler_helper/metric

### PR DESCRIPTION
**Pull request by nighood**
## Check List

- [ ] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: ding/league/metric.py [hunk 1]
```diff
@@ -18,6 +29,20 @@ def get_new_rating(cls,
...
+
+        Arguments:
+            - rating_a (:obj:`int`): The current rating of player A.
+            - rating_b (:obj:`int`): The current rating of player B.
+            - result (:obj:`int`): The result of the game: 1 for player A win, 0 for draw, -1 for player B win.
+            - k_factor (:obj:`int`): The K-factor used in the Elo rating system. Defaults to 32.
+            - beta (:obj:`int`): The beta value used in the Elo rating system. Defaults to 200.
+
+        Returns:
+            -ret (:obj:`Tuple[int, int]`): The new ratings for player A and player B, respectively.
```
Comment (kxzxvbk) [inline ding/league/metric.py:44]: There should be a blank between '-' and 'ret'

File: ding/league/metric.py [hunk 2]
```diff
@@ -35,10 +60,25 @@ def get_new_rating_array(
...
+        Arguments:
+            - rating (obj:`np.ndarray`): An array of current ratings for each player.
+            - result (obj:`np.ndarray`): An array of game results, where 1 represents a win, 0 represents a draw, \
+                and -1 represents a loss.
+            - game_count (obj:`np.ndarray`): An array of game counts for each player.
+            - k_factor (obj:`int`): The K-factor used in the Elo rating system. Defaults to 32.
+            - beta (obj:`int`): The beta value used in the Elo rating system. Defaults to 200.
+
+        Returns:
+            -ret(obj:`np.ndarray`): An array of new ratings for each player.
```
Comment (kxzxvbk) [inline ding/league/metric.py:76]: Also the blank

File: ding/league/metric.py [hunk 3]
```diff
@@ -62,14 +109,33 @@ def __repr__(self) -> str:
...
+        Arguments:
+            - mu (:obj:`float`): The mean value of the player's skill rating. If not provided, the default \
+                TrueSkill mean is used.
+            - sigma (:obj:`float`): The standard deviation of the player's skill rating. If not provided, \
+                the default TrueSkill sigma is used.
+            - elo_init (:obj:int`): The initial Elo rating value for the player. If not provided, the default \
+                elo_init value of the LeagueMetricEnv class is used.
+
+        Returns:
+            - PlayerRating: A player rating object with the specified mean, standard deviation, and Elo rating.
```
Comment (kxzxvbk) [inline ding/league/metric.py:137]: The return should also contain typing (:obj:`PlayerRating`)

File: ding/league/metric.py [hunk 4]
```diff
@@ -62,14 +109,33 @@ def __repr__(self) -> str:
...
+        """
+        Overview:
+            Creates a new player rating object with the specified mean, standard deviation, and Elo rating.
+
+        Arguments:
+            - mu (:obj:`float`): The mean value of the player's skill rating. If not provided, the default \
+                TrueSkill mean is used.
+            - sigma (:obj:`float`): The standard deviation of the player's skill rating. If not provided, \
+                the default TrueSkill sigma is used.
+            - elo_init (:obj:int`): The initial Elo rating value for the player. If not provided, the default \
```
Comment (kxzxvbk) [inline ding/league/metric.py:133]: (:obj:`int`)

File: ding/league/metric.py [hunk 5]
```diff
@@ -111,6 +181,19 @@ def rate_1vs1(self,
...
+            Rates a team of players against a single player in a 1 vs C match and returns the updated rating \
+            for the team.
+
+        Arguments:
+            - team1 (:obj:`PlayerRating`): The rating object representing the team of players.
+            - team2 (:obj:`PlayerRating`): The rating object representing the single player.
+            - result (:obj:`List[str]`): The result of the match. Can be 'wins', 'draws', or 'losses'.
+
+        Returns:
+            - PlayerRating: The updated rating for the team of players.
```
Comment (kxzxvbk) [inline ding/league/metric.py:195]: Type should be added.

File: ding/utils/data/collate_fn.py [hunk 6]
```diff
@@ -18,6 +18,20 @@
 
 
 def ttorch_collate(x, json: bool = False, cat_1dim: bool = True):
```
Comment (kxzxvbk) [inline ding/utils/data/collate_fn.py:20]: The return type should be added, also for the comments below.

File: ding/utils/data/collate_fn.py [hunk 7]
```diff
@@ -124,17 +141,27 @@ def default_collate(batch: Sequence,
...
+        >>> batch = [
+                {'data0': [torch.tensor([1, 2, 3]), torch.tensor([4, 5, 6])]}, 
+                {'data1': [torch.tensor([7, 8, 9]), torch.tensor([10, 11, 12])]}
+            ]
+        >>> collated_data = timestep_collate(batch)
+        >>> print(collated_data['data'].shape)
+        torch.Size([2, 2, 3])
     """
 
     def stack(data):
```
Comment (kxzxvbk) [inline ding/utils/data/collate_fn.py:167]: Why there is no comment for this function?

File: ding/utils/profiler_helper.py [hunk 8]
```diff
@@ -36,6 +58,14 @@ def write_profile(self, pr, folder_path):
         pr.dump_stats(folder_path + "/profile.prof")
 
     def profile(self, folder_path="./tmp"):
```
Comment (kxzxvbk) [inline ding/utils/profiler_helper.py:60]: folder_path: str = "./tmp"

File: ding/utils/data/collate_fn.py [hunk 9]
```diff
@@ -18,6 +18,20 @@
...
+
+    Arguments:
+        - x : The input list of tensors or nested dictionaries of tensors.
+        - json (:obj:`bool`): If True, converts the output to JSON format. Defaults to False.
+        - cat_1dim (:obj:`bool`): If True, concatenates tensors with shape (B, 1) along the last dimension. \
+            Defaults to True.
+
+    Returns:
+        The collated output tensor or nested dictionary of tensors.
+    """
```
Comment (PaParaZz1) [inline ding/utils/data/collate_fn.py:34]: add an example for this function

File: ding/utils/profiler_helper.py [hunk 10]
```diff
@@ -10,15 +10,37 @@ def register_profiler(write_profile, pr, folder_path):
...
+            Create a directory if it doesn't exist.
+
+        Arguments:
+            - directory (:obj:`str`): The path of the directory to be created.
+        """
         if not os.path.exists(directory):
             os.makedirs(directory)
 
-    def write_profile(self, pr, folder_path):
+    def write_profile(self, pr, folder_path: str):
```
Comment (PaParaZz1) [inline ding/utils/profiler_helper.py:35]: add typing lint for `pr`

File: ding/utils/profiler_helper.py [hunk 11]
```diff
@@ -36,6 +58,14 @@ def write_profile(self, pr, folder_path):
         pr.dump_stats(folder_path + "/profile.prof")
 
     def profile(self, folder_path="./tmp"):
+        """
+        OverView:
+            Enable profiling and save the results to files.
+
+        Arguments:
+            - folder_path (str, optional): The path of the folder where the profiling files will be saved. \
```
Comment (PaParaZz1) [inline ding/utils/profiler_helper.py:66]: modify typing lint

File: ding/league/metric.py [hunk 12]
```diff
@@ -80,22 +146,26 @@ def create_rating(self, mu: float = None, sigma: float = None, elo_init: int = N
 
     @staticmethod
     def _rate_1vs1(t1, t2, **kwargs):
-        t1_elo, t2_elo = t1.elo, t2.elo
```
Comment (PaParaZz1) [inline ding/league/metric.py:83]: why delete these codes

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]
